torch/csrc/jit/passes/quantization.cpp

```diff
@@ -676,5 +676,246 @@ TORCH_API script::Module InsertObservers(
+                          "getQParam expects the corresponding observer for ",
+                          v->debugName(),
+                          " exists.");
+    auto observer_module = module_.find_module(observer_name.value());
+    TORCH_INTERNAL_ASSERT(observer_module,
+                          "getQParam expects the corresponding observer for ",
+                          v->debugName(),
+                          " exists.");
+    auto calculate_qparams = observer_module.value().get_method("calculate_qparams");
+    IValue qparams = calculate_qparams(std::vector<IValue>());
+    return qparams;
+}
+
+void QuantizeHelper::quantizeBias(Value* v) {
+  // Traverse to the place where this is used
+  Node* aten_call_node = nullptr;
+  std::vector<std::string> ops_with_bias = {"aten::conv2d", "aten::_convolution"};
+  for (const auto& use: v->uses()) {
+    if (std::find(ops_with_bias.begin(), ops_with_bias.end(),
+                  use.user->kind().toQualString()) != ops_with_bias.end()) {
+      // Make sure there is no observer module for bias
+      c10::optional<script::Module> observer_module;
+      auto observer_name = findObserverName(v);
+      TORCH_INTERNAL_ASSERT(!observer_name,
+                            "bias should not be observed!");
+      Value* activation = use.user->inputs()[0];
+      Value* weight = use.user->inputs()[1];
+      // Get qparam from activation
+      IValue act_qparam = getQParam(activation);
+      // Get qparam from weight
+      IValue weight_qparam = getQParam(weight);
+      IValue bias_scale =  at::scalar_tensor(
+          c10::Scalar(act_qparam.toTuple()->elements()[0].toTensor().item().toDouble() *
+                      weight_qparam.toTuple()->elements()[0].toTensor().item().toDouble()),
+          at::kDouble);
+      IValue bias_qparam = c10::ivalue::Tuple::create(
+          std::vector<IValue>({bias_scale,
+                               at::scalar_tensor(c10::Scalar(0))}), act_qparam.toTuple()->type);
+      Node* dequant = insertQuantDeQuantCall(v, bias_qparam, at::kQInt32);
+      v->replaceAllUsesWith(dequant->output());
+      Node* q = traverseToQuantNode(dequant);
+      TORCH_INTERNAL_ASSERT(q != nullptr);
+      q->replaceInputWith(dequant->output(), v);
+      break;
+    }
+  }
+}
+
+void QuantizeHelper::quantizeTensor(Value* v,
+                                    bool insert_after) {
+  auto observer_name = findObserverName(v);
+  if (!observer_name) {
+    return;
+  }
+  auto observer_module = module_.find_module(observer_name.value());
+  TORCH_INTERNAL_ASSERT(observer_module,
+                        "Expect observer module " +                     \
+                        observer_name.value() + " in quantizeTensor");
+  // remove observer_module
+  observer_modules_to_remove_.push_back(observer_name.value());
+  // remove observer forward call
+  for (const Use& u: v->uses()) {
+    Node* user = u.user;
+    if (user->kind() == prim::CallMethod &&
```

Comment:
This check can be factored out to a function `isObserverForwardCall` (or something like that), which can be used here and in `findObserverName`.

Reply:
in `findObserverName` we actually don't know the name yet, I think the only check we can extract is `user->kind() == prim::CallMethod && user->s(attr::name) == "forward"` probably not worthwhile to do..